private/pkg/manifest/blob.go

```diff
@@ -97,69 +101,31 @@ type BlobSet struct {
 	digestToBlob map[string]Blob
 }
 
-type blobSetOptions struct {
-	validateContent bool
-	skipNilBlobs    bool
```

Comment:
Both of these options should have probably defaulted to true all along, and there are no callers that are negatively affected by setting these to true so I'm simplifying the API surface area and removing the options.